src/lib/update-lock.ts

```diff
@@ -42,7 +42,11 @@ function lockFilesOnHost(appId: number, serviceName: string): string[] {
  * prevent reboot. If the Supervisor reboots while those services are still running,
  * the device may become stuck in an invalid state during HUP.
  */
-export function ensureNoHUPBreadcrumbsOnHost(): Promise<boolean | never> {
+export function ensureNoHUPBreadcrumbsOnHost({
```

Comment:
Can we rename this function to something like `foundHUPBreadcrumbsOnHost` to avoid any confusion around the `ensureNo` prefix?

Reply:
I also don't love the name. I don't think it explains well what the function does, particularly since you cannot infer from the name and reading the code that the function will throw if the condition is not met. 

I think it would make more sense to make the signature more similar to that of the `lock` function (or the `lockIfNecessary` function), where the callback doesn't get executed if the condition is not met. I'm OK with not doing that if the name reflects the behavior though.

Reply:
How about `lockDuringHUP`?

Reply:
@pipex You said that it'd be better if this lock functionality had a signature similar to updateLock.lock. This function was a part of updateLock.lock originally, but because the lock doesn't work when there's no application on the device, I separated this functionality out. If I wanted to keep HUP breadcrumb locking functionality within updateLock.lock, I could do so by allowing the appId paramter to be optional, and accounting for different types of locking conditions in updateLock,lock. This would be a bit more of a refactor than than the HUP breadcrumbs PR that was already merged, so I decided against it.

That being said, I can take a look at a refactor if you and @20k-ultra think it's something that would simplify the code.

Reply:
I think the refactor is outside scope for the PR, but eventually we should cleanup locking behavior on the codebase. For the function name. Maybe something like `abortIfHUPInProgress`? I want it to make clear that the operation will be interrupted (through an exception) if there is HUP.

Reply:
Sounds good. I'll just change the name then, but we can make a note to look at simplifying the lock code at a future time